spec/Section 3 -- Type System.md

```diff
@@ -1072,6 +1134,42 @@ Interface types have the potential to be invalid if incorrectly defined.
          characters {"__"} (two underscores).
       2. The argument must accept a type where {IsInputType(argumentType)}
          returns {true}.
+3. An interface type may declare that it implements one or more unique
+   interfaces, but may not implement itself.
+4. An interface type must be a super-set of all interfaces it implements:
+   1. The implementing interface type must include a field of the same name for
+   every field defined in an implemented interface.
+      1. The implementing interface field must be of a type which is equal to or
+         a sub-type of the implemented interface field (covariant).
+         1. An implementing interface field type is a valid sub-type if it is
+            equal to (the same type as) the implemented interface field type.
+         2. An implementing interface field type is a valid sub-type if it is an
+            Object type and the implemented interface field type is a Union type
+            and the implementing interface field type is a possible type of the
+            implemented interface field type.
+         3. An implementing interface field type is a valid sub-type if it is an
+            Object or Interface type and the implemented interface field type is
+            an Interface type and the implementing interface field type
+            implements the implemented interface field type.
+         4. An implementing interface field type is a valid sub-type if it is a
+            List type and the implemented interface field type is also a List
+            type and the list-item type of the implementing interface field type
+            is a valid sub-type of the list-item type of the implemented
+            interface field type.
+         5. An implementing interface field type is a valid sub-type if it is a
+            Non-Null variant of a valid sub-type of the implemented interface
+            field type.
+      2. The implementing interface field must include an argument of the same
+         name for every argument defined in the implemented interface field.
+         1. The implementing interface field argument must accept the same type
+            (invariant) as the implemented interface field argument.
+      3. The implementing interface field may include additional arguments not
+         defined in the implemented interface field, but any additional argument
+         must not be required.
```

Comment:
I may make an editorial change here to factor this out into an algorithm so that it can exactly mirror the same rules defined by object